kairos-prover/kairos-prover-risc0-server/src/main.rs

```diff
@@ -0,0 +1,200 @@
+use std::time::Instant;
+
+use axum::{http::StatusCode, Json};
+use axum_extra::routing::{RouterExt, TypedPath};
+use kairos_circuit_logic::{ProofInputs, ProofOutputs};
+
+// These constants represent the RISC-V ELF and the image ID generated by risc0-build.
+// The ELF is used for proving and the ID is used for verification.
+use methods::{PROVE_BATCH_ELF, PROVE_BATCH_ID};
+use risc0_zkvm::{ExecutorEnv, Receipt};
+
+#[tokio::main]
+async fn main() {
+    let _ = dotenvy::dotenv();
+    // Initialize tracing. In order to view logs, run `RUST_LOG=info cargo run`
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::filter::EnvFilter::from_default_env())
+        .init();
+
+    let socket_addr = std::env::var("KAIROS_PROVER_SERVER_SOCKET_ADDR")
+        .expect("Failed to fetch environment variable KAIROS_PROVER_SERVER_SOCKET_ADDR");
+    let socket_addr = socket_addr
+        .parse::<std::net::SocketAddr>()
+        .expect("Failed to parse KAIROS_SERVER_SOCKET_ADDR");
+
+    let app = axum::Router::new()
+        .typed_post(prove_batch_route)
+        .with_state(());
+
+    tracing::info!("starting http server on `{}`", socket_addr);
+    let listener = tokio::net::TcpListener::bind(socket_addr).await.unwrap();
+    tracing::info!("listening on `{}`", socket_addr);
+    axum::serve(listener, app).await.unwrap()
+}
+
+#[derive(TypedPath, Debug, Clone, Copy)]
+#[typed_path("/api/v1/prove/batch")]
+pub struct ProveBatch;
+
+pub async fn prove_batch_route(
+    _: ProveBatch,
```

Comment:
```
#[derive(TypedPath, Debug, Clone, Copy)]
#[typed_path("/api/v1/prove/batch")]
pub struct ProveBatch;
```

This is the typed path, since the endpoint takes JSON, the there is no data from the URL, hence an empty struct.